### PR TITLE
More Tests

### LINKED ISSUE #14: Setup CI
Lets get some continuous integration setup once the tests are in a better condition, because that is a good thing to do. I am thinking TravisCI.

### PR DESCRIPTION
**Pull request by damonkelley**
Will ~~complete~~ satisfy #3 once completed.

Task List:
- [x] All `TODO`s have been taken care of.
- [x] ~~No~~ Few expected failures (via `@pytest.mark.xfail`). 
- [ ] Tests should have docstrings if they need further clarification

Since this branch is to _add_ more tests, not to _pass_ more tests, I will leave any expected failures that require a change to name module.

**Edit**:
This will also close #14

### CODE REVIEW

File: tests/test_models.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import pytest
+from name import models
+
+
+class TestIdentifier_Type:
+
```
Comment (damonkelley) [inline tests/test_models.py]: Remove this whitespace.

**Comment by damonkelley** [comment]
A few `TODO`s to take care of, and this will be ready.

**Comment by damonkelley** [comment]
As of 2acca85, our test output is the following:

``` sh
71 passed, 7 xfailed, 1 xpassed in 8.71 seconds
```

**Comment by damonkelley** [comment]
Once there are tests for the decorators, this should be ready for merging. There will still be some failures and TODOs to take care of, but those are more suited for separate issues.

**Comment by damonkelley** [comment]
@ldko, this is should be ready. Would you mind giving it a once over, and let me know if you notice anything fishy? 

There are still a few tests that are marked with `@pytest.mark.xfail`, that will be implemented/changed once some other issues are taken care of.

File: tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    '''Twenty Name objects.
+
+    The name type for each object is randomly determined.
+    This fixture is good for testing search functionality, or
+    any occasion where you need a large amount of Names and
+    the type of those Names is arbitrary.
+    '''
+    for x in range(21):
+        models.Name.objects.create(
+            name="Name {}".format(x),
```
Comment (ldko) [inline tests/conftest.py]: need a positional argument in the brackets if we want to support python 2.6 in places like line 39 of tests/conftest.py and in a handful of other places in the changes

**Comment by ldko** [comment]
only other thing was many tests without docstrings

**Comment by damonkelley** [comment]
Good call @ldko.

File: tests/conftest.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        models.Name.objects.create(
+            name="Personal {}".format(x), name_type=0)
+        models.Name.objects.create(
+            name="Organization {}".format(x), name_type=1)
+        models.Name.objects.create(
+            name="Event {}".format(x), name_type=2)
+        models.Name.objects.create(
+            name="Software {}".format(x), name_type=3)
+        models.Name.objects.create(
+            name="Building {}".format(x), name_type=4)
```
Comment (damonkelley) [inline tests/conftest.py]: `seach_fixtures` also needs positional arguments in brackets.

File: tests/test_search.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    response = client.get(
+        reverse('name_search') + query)
+
+    assert response.status_code is 200
+
+
+def test_search_with_q(client, twenty_name_fixtures):
+    '''Search with q only. No name_types provided'''
+    name = twenty_name_fixtures.first()
+    url = reverse('name_search') + "?q={}".format(name.name)
```
Comment (damonkelley) [inline tests/test_search.py]: need a positional argument in brackets here too for Python 2.6

**Comment by damonkelley** [comment]
Even if we don't end up supporting Python 2.6, we should keep the string formatting consistent, using `{0}` over `{}`

**Comment by damonkelley** [comment]
We have also bundled #11 into this as well.

**Comment by damonkelley** [comment]
I have been wondering if the Travis build would succeed or not if there were PEP8 errors. 6ef1ab2 confirms that it will fail. very cool :+1:

**Pull request merged**